tensorflow/python/ops/image_ops_impl.py

```diff
@@ -110,6 +110,26 @@ def _ImageDimensions(image):
             for s, d in zip(static_shape, dynamic_shape)]
 
 
+def _BatchImageDimensions(image):
```

Comment:
There is a lot of code duplication between this function and `_ImageDimensions(image)` above.
Can you please unify them by adding a `rank` argument to `_ImageDimensions()`?
Then you can call it with rank=3 or rank=4.